quinn-proto/src/connection/datagrams.rs

```diff
@@ -149,10 +149,28 @@ impl DatagramState {
         Ok(was_empty)
     }
 
-    pub(super) fn write(&mut self, buf: &mut Vec<u8>, max_size: usize) -> bool {
-        let datagram = match self.outgoing.pop_front() {
-            Some(x) => x,
-            None => return false,
+    pub(super) fn write(
+        &mut self,
+        buf: &mut Vec<u8>,
```

Comment:
Can you add a docstring that elucidates the distinction between `max_size` and `drop_above_size`?

Reply:
I've instead rewritten this logic to proactively drop all oversized datagrams when a black hole is detected, rather than checking them lazily. It's a cosmetic difference for purposes of this PR, but it preempts a bug in #1837 where datagrams could become unrecoverably clogged by an over-MTU datagram due to `can_send_1rtt` skipping datagrams when an over-segment-size datagram is at the front of the queue, before packet assembly takes place.

Reply:
Makes sense.